app/src/main/java/org/simple/clinic/widgets/PatientSearchResultItemView.kt

```diff
@@ -91,13 +95,41 @@ class PatientSearchResultItemView(
     addressLabel.text = address.completeAddress
   }
 
-  private fun renderPatientNameAgeAndGender(fullName: String, gender: Gender, dateOfBirth: DateOfBirth) {
+  private fun highlightPatientName(searchQuery: String, patientSearchResult: PatientSearchResultViewModel, dateOfBirth: DateOfBirth): Name {
+    val canHighlight = searchQuery.isNotBlank() && patientSearchResult.fullName.contains(searchQuery, ignoreCase = true)
+    return if (canHighlight) {
+      Name.Highlighted(
+          patientName = renderPatient(patientSearchResult.fullName, patientSearchResult.gender, dateOfBirth),
+          highlightStart = patientSearchResult.fullName.indexOf(searchQuery, ignoreCase = true),
+          highlightEnd = patientSearchResult.fullName.indexOf(searchQuery, ignoreCase = true) + searchQuery.length)
+    } else {
+      Name.Plain(renderPatient(patientSearchResult.fullName, patientSearchResult.gender, dateOfBirth))
+    }
+  }
+
+  private fun renderPatientNameAgeAndGender(name: Name) {
+    when (name) {
+      is Name.Highlighted -> {
+        val highlightName = SpannableStringBuilder(name.patientName)
+        val highlightColor = ContextCompat.getColor(context, R.color.facility_search_query_highlight)
+        highlightName.setSpan(BackgroundColorSpan(highlightColor), name.highlightStart, name.highlightEnd, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE)
+        patientNameAgeGenderLabel.text = highlightName
+      }
+      is Name.Plain -> {
+        patientNameAgeGenderLabel.text = name.patientName
+      }
+    }
+  }
+
+  private fun renderPatient(fullName: String, gender: Gender, dateOfBirth: DateOfBirth): String {
```

Comment:
Here we are not actually rendering the patient information, we are just getting the patient name, age, and dob string. Can we rename this to represent that?